### ISSUE TITLE
variable ownerType is already defined in class com.onshape.api.requests.MetadataCategoryUpdatePropertyRequest.Builder

### ISSUE THREAD

**Issue by ethan92429** [onshape-public/java-client#28]
I am trying to generate java-client from demo-c today, as we want updated java-client with some fixes recently submitted. The java-client generation is failing with below error. Could you please take a look?
 [INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  22.730 s
[INFO] Finished at: 2020-07-30T10:58:23+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project api-client: Compilation failure: Compilation failure:
[ERROR] /home/ccsteam/repos/api-generator-from-chris/api-generator-master/onshape-java/api-generator/target/java-build/src/main/java/com/onshape/api/requests/MetadataCategoryUpdatePropertyRequest.java:[354,20] variable ownerType is already defined in class com.onshape.api.requests.MetadataCategoryUpdatePropertyRequest.Builder (edited)

**Comment by harmanpa** [comment]
@ethan92429 What is the endpoint json for the metadata-category-update-property method?

**Comment by ethan92429** [comment]
```
{
        "type": "delete",
        "url": "/metadatacategory/property/:pid",
        "title": "Delete metadata schema property",
        "name": "deleteProperty",
        "description": "Deletes a metadata category property",
        "group": "MetadataCategory",
        "version": "1.0.0",
        "permission": [
          {
            "name": "internal:undocumented"
          },
          {
            "name": "OAuth2Deny"
          }
        ],
        "parameter": {
          "fields": {
            "QueryParam": [
              {
                "group": "QueryParam",
                "type": "String",
                "optional": false,
                "field": "Requesting",
                "description": "owner id, required"
              },
              {
                "group": "QueryParam",
                "type": "Number",
                "optional": true,
                "field": "ownerType",
                "description": "Owner type, optional that can be: 1:company 2:onshape (default: 1)"
              }
            ],
            "PathParam": [
              {
                "group": "PathParam",
                "type": "String",
                "optional": false,
                "field": "pid",
                "description": "Property ID"
              }
            ]
          }
        },
        "groupTitle": "MetadataCategory"
      }
```

**Comment by ethan92429** [comment]
As you can see, `ownerType` is defined exactly once as a queryParam, as a body param, and as a response param.

**Comment by harmanpa** [comment]
Do the query param and body param have different meanings?

**Comment by ethan92429** [comment]
No - but they have to agree. Not the best design, but that is what we've got!

**Comment by harmanpa** [comment]
The offending line of code is here:
https://github.com/onshape-public/api-generator/blob/6762b20d0e9b331ba853c5ad6a73bac19135b405/onshape-java/api-generator/src/main/java/com/onshape/api/generator/java/JavaEndpointTarget.java#L298

If we can check that the field (i.e. ownerType) also exists as a body parameter at this point then we simply don't try to add another field to the Java class. The client behaviour will then be to use the same value as both a body param and query param (so no need to check in client they are same as they would use the same field).

**Comment by ethan92429** [comment]
I don't think the client should make the assumption that query params and body params should necessarily be the same? What do you think?

**Comment by harmanpa** [comment]
I guess it depends if this (having params of different types with same name) is a one-off or a common pattern?

**Comment by ethan92429** [comment]
Unfortunately we do this commonly, but I don't want to make an assumption that they are supposed to be the same... Our API seems to have a habit of not following patterns too well, so there may be some endpoints that actually have different behavior depending on which (body or query) param is changed.

**Comment by harmanpa** [comment]
OK, another option is to prefix/suffix the name of either affected query params or all query params (e.g. "queryOwnerType" / "ownerTypeQuery") so there isn't a clash?

**Comment by ethan92429** [comment]
Yes - I see that the alternative isn't that clean either. OK - for now, let's go with your original suggestion. If we come across a time when the names are the same but the behavior is different, that will probably be a good thing since it will alert us to unexpected behavior in our API.
